### PR TITLE
Implement project status completed to Projects table and its related modifications with project start and end date

### PR DESCRIPTION
**Pull request by Humam-Yousif**

### CODE REVIEW

**Review by KAJOiq — APPROVED** [review]

File: Controllers/Mobile/v1/ProjectsController.cs [hunk 1]
```diff
@@ -185,8 +185,8 @@ public async Task<IActionResult> SubmitBidAsync(long projectId, [FromBody] BidIn
             Project? storedProject = await projectService.FindProjectWithBidsAsync(projectId);
             if (storedProject == null)
                 return NotFound(CreateErrorResponse(StatusCodes.Status404NotFound.ToString(), "Project not found."));
-            if (storedProject.Status != Constants.PROJECT_STATUS_AVAILABLE)
-                return Conflict(CreateErrorResponse(StatusCodes.Status409Conflict.ToString(), "Cannot submit a bid for project that is not available for bids."));
+            if (storedProject.Status != Constants.PROJECT_STATUS_PENDING)
+                return Conflict(CreateErrorResponse(StatusCodes.Status409Conflict.ToString(), "Cannot submit a bid for project that is not Pending for bids."));
```
Comment (diyar214) [inline Controllers/Mobile/v1/ProjectsController.cs:189]: "Cannot submit a bid for project that is not Pending for bids."
Add `that is not Pending or Completed`

**Review by diyar214 — COMMENTED** [review]
Changes needs to be discussed and then tested.

File: Controllers/Mobile/v1/ProfileController.cs [hunk 2]
```diff
@@ -72,7 +72,7 @@ public async Task<IActionResult> GetUserStatisticsAsync()
                                                                   TasksStatisticsDTO.FromTasks(storedTasks))));
         }
         [HttpGet("projects")]
-        public async Task<IActionResult> GetProjectsForUserDashboard([AllowedValues(PROJECT_STATUS_AVAILABLE, PROJECT_STATUS_CLOSED, PROJECT_STATUS_COMPLETED)] string status = "",
+        public async Task<IActionResult> GetProjectsForUserDashboard([AllowedValues(PROJECT_STATUS_PENDING, PROJECT_STATUS_IN_PROGRESS, PROJECT_STATUS_COMPLETED)] string status = "",
```
[the remark this answers is not in the document]
Comment (Humam-Yousif) [reply Controllers/Mobile/v1/ProfileController.cs:75]: yeah ;(

File: Controllers/Mobile/v1/ProjectsController.cs (hunk 1 above)
Comment (Humam-Yousif) [reply Controllers/Mobile/v1/ProjectsController.cs]: done !

File: Controllers/Mobile/v1/ProjectsController.cs [hunk 3]
```diff
@@ -528,5 +528,31 @@ public async Task<IActionResult> GetProjectCommentsAsync([FromRoute] long projec
...
 
+        [Authorize(Roles = Constants.USER_TYPE_CLIENT)]
+        [HttpPatch("{projectId}/completed")]
+        public async Task<IActionResult> MarkProjectAsCompletedAsync([FromRoute] long projectId)
+        {
+            if (!ModelState.IsValid)
+                return CustomBadRequest();
+
+            long authenticatedClientId = authService.GetUserId((ClaimsIdentity)HttpContext.User.Identity);
+            string nameOfAuthenticatedClient = authService.GetNameOfUser((ClaimsIdentity)HttpContext.User.Identity);
```
Comment (diyar214) [inline Controllers/Mobile/v1/ProjectsController.cs:539]: I don't see the use of this variable, please explain it to me. Also, if you need the nameOfAuthenticatedClient you  can get the name of the client from the authenticatedUser variable you have defined below.

**Review by diyar214 — COMMENTED** [review]
Kindly requesting to get feedback on t below comments.

**Review by diyar214 — APPROVED** [review]
LGTM!

**Review by diyar214 — APPROVED** [review]
All good

**Pull request merged**